### ISSUE TITLE
Hyper-parameter for VPT

### ISSUE THREAD

**Issue by Charleshhy** [ShoufaChen/AdaptFormer#6]
Thanks a lot for the great work! I am having trouble re-producing the results for VPT [46], so I wonder whether it's possible to share your hyper-parameters for training VPT that are reported in Table 5 under the supervised pre-training?
<img width="1205" alt="image" src="https://user-images.githubusercontent.com/39027479/172750569-142052ec-3b70-4ee8-8b50-535dae619881.png">

Kind regards,
Haoyu

**Comment by ShoufaChen** [comment]
Hi, Thanks for your interest.

We use strictly the same training recipe for all fine-tuning methods. The training code for the video is released.  For image, please refer to https://github.com/ShoufaChen/AdaptFormer/issues/2.

Would you mind sharing your reproducing performance? higher or lower than ours?

**Comment by Charleshhy** [comment]
Hi Shuofa,

Thanks for your prompt reply :)! Can you be more specific? E.g., Are you using 1 prompt for CIFAR-100, SVHN, and Food-101? Is the final lr for 1024 batch-size 0.4 according to [this line](https://github.com/ShoufaChen/AdaptFormer/blob/d8c95b51337feb5109a214e5b99e8f44bd3371d1/main_video.py#L364)? Are the prompts initialized with Xavier uniform initialization?

Many thanks,
Haoyu

**Comment by ShoufaChen** [comment]
(1) Yes, the number of prompt is 1 for image datasets;  (2) The real LR is 0.4 according to the linear scaling rue. However, We found it is not stable in cases, especially for Swin Transformer. As mentioned in https://github.com/ShoufaChen/AdaptFormer/issues/2, we found smaller lr can bring further gains for all fine-tuning methods. (3). Prompts are initialized by 
```
for eee in self.embeddings:
    torch.nn.init.xavier_uniform_(eee.data)
```
for both video and image. 

It would be appreciated if you could share your performance.

**Comment by Charleshhy** [comment]
Many thanks! 

Currently, I can only achieve 90.4% on CIFAR100 while [this paper](https://arxiv.org/abs/2203.15243#:~:text=In%20this%20paper%20we%20propose,capabilities%20on%20previously%20learned%20tasks.) achieves 91.1% top-1 accuracy. I will let you know how it goes with your provided specifications! I will close this issue for now :)

Kind regards,
Haoyu
